sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridPublisherClientBuilder.java

```diff
@@ -254,11 +249,19 @@ public EventGridPublisherClientBuilder credential(AzureSasCredential credential)
 
     /**
      * Set the domain or topic endpoint. This is the address to publish events to.
+     * It must be the full url of the endpoint instead of just the hostname.
      * @param endpoint the endpoint as a url.
      *
      * @return the builder itself.
+     * @throws NullPointerException if {@code endpoint} is null.
+     * @throws IllegalArgumentException if {@code endpoint} cannot be parsed into a valid URL.
      */
     public EventGridPublisherClientBuilder endpoint(String endpoint) {
+        try {
+            new URL(Objects.requireNonNull(endpoint, "'endpoint' cannot be null."));
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsWarning(new IllegalArgumentException("'endpoint' must be a valid URL"));
```

Comment:
Include the root exception when creating a new exception instance.
```suggestion
            throw logger.logExceptionAsWarning(new IllegalArgumentException("'endpoint' must be a valid URL", ex));
```